### PR TITLE
Account for null value in redux-routine createRuntime

### PR DESCRIPTION
**Pull request by nerrad**
## Description

In the `createRuntime` method found in `packages/redux-routine/src/runtime`, the following conditional would produce an error if `value` is null:

```
( typeof value !== 'object' || value.type !== actionType )
```

It appears that `null` could be an incoming value and since `typeof null === 'object'` evaluates to `true` there would be an error with the following `value.type` test.

In this pull:

- created `isAction` and `isSpecificAction` for detection objects that serve as action objects.
- implemented the above two functions in `createRuntime` to explicitly check that value "quacks like an action" and "quacks like the expected action".

## How has this been tested?

- I've added unit tests for the new utility functions and verified they passed.

## Types of changes

Bug fix.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
LGTM 👍 I'd appreciate a "naming" check from @aduth or @tofumatt for the function names as I'm very bad at it :)

**Comment by aduth** [comment]
I might prefer `isActionOfType` over `isSpecificAction` just because it does better at signally what _specifically_ it's testing for, but all things considered it's a small thing.

**Comment by nerrad** [comment]
> I might prefer isActionOfType over isSpecificAction

Ya I like that name suggestion too, will change.

File: packages/redux-routine/src/is-action.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+export function isAction( object ) {
+	return isPlainObject( object ) && isString( object.type );
+}
+
+/**
+ * Returns true if the given object quacks like an action and has a specific
+ * action type
+ *
+ * @param {*} object Object to test
```
Comment (aduth) [inline packages/redux-routine/src/is-action.js]: Nit: Should align the doc tag fragments

https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#functions

```suggestion
 * @param {*}      object       Object to test
```

**Pull request merged**